lang/python/CMakeLists.txt

```diff
@@ -0,0 +1,130 @@
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#  All rights reserved.
+#
+# See the file LICENSE for redistribution information.
+#
+
+project(python_swig C)
+
+# UseSWIG: Policy option to use the standard target name over a generated name.
+if(POLICY CMP0078)
+    cmake_policy(SET CMP0078 NEW)
+endif()
+
+# UseSWIG: Policy option to honor SWIG_MODULE_NAME.
+if(POLICY CMP0086)
+    cmake_policy(SET CMP0086 NEW)
+endif()
+
+find_package(SWIG REQUIRED)
+include(${SWIG_USE_FILE})
+
+set(python_libs)
+set(python_version)
+if("${CMAKE_VERSION}" VERSION_LESS_EQUAL "3.12.0")
+    # This method of finding python libs has been deprecated since version 3.12.
+    # If we are running with a greater CMake version, opt to use the Python3 package.
+    set(Python_ADDITIONAL_VERSIONS 3.9 3.8 3.7 3.6 3.5)
+    find_package(PythonInterp REQUIRED)
+    find_package(PythonLibs REQUIRED)
+    include_directories(${PYTHON_INCLUDE_DIRS})
+    set(python_libs ${PYTHON_LIBRARIES})
+    set(python_version ${PYTHON_VERSION_STRING})
+else()
+    find_package(Python3 COMPONENTS Interpreter Development REQUIRED)
+    include_directories(${Python3_INCLUDE_DIRS})
+    set(python_libs ${Python3_LIBRARIES})
+    set(python_version ${Python3_VERSION})
+endif()
+
+# Manually specify the interface name '_wiredtiger' over a generated name (for backwards compatibility).
```

Comment:
Just curious what 'backward compatibility means here - is it related to different CMake versions, or CMake vs autotool?

Reply:
Sorry for the confusing comment. Backwards compatibility here refers to our python module wrappers expecting a generated swig library name of `_wiredtiger` when importing it (which our autoconf build produces). I've expanded the comment a bit to make this clearer.